test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/rest/ResteasyClientBuilderFactoryTest.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.services.test.utils.core.rest;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.nullValue;
+
+import org.apache.http.HttpHost;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.engines.ApacheHttpClient4Engine;
+import org.junit.Test;
+
+public class ResteasyClientBuilderFactoryTest {
+
+    private static final String PROPERTY_HTTP_PROXY_HOST = "http.proxyHost";
+    private static final String PROPERTY_HTTP_PROXY_PORT = "http.proxyPort";
+    private static final String PROXY_HOST = "proxy.moj.com";
+    private static final int PROXY_PORT = 3125;
+
+    @Test
+    public void shouldSetProxyIfConfigured() throws Exception {
+        givenProxyIsConfigured();
+
+        final ResteasyClient client = ResteasyClientBuilderFactory.clientBuilder().build();
+        final HttpHost actualProxySettings = ((ApacheHttpClient4Engine) client.httpEngine()).getDefaultProxy();
+
+        assertThat(actualProxySettings.getHostName(), is(PROXY_HOST));
+        assertThat(actualProxySettings.getPort(), is(PROXY_PORT));
+
+        clearProxyConfiguration();
+    }
+
+    @Test
+    public void shouldNotSetProxyIfNotConfigured() throws Exception {
+        givenProxyIsNotConfigured();
```

Comment:
Why are we testing that no proxy is set? We might want to ensure that before the test is run that the properties are clear, but it's entirely possible that the tests might be run on an environment where the system properties **are** set, because they are genuinely needed.

If these tests need the values to be clear, I suggest you grab the existing values before the test, clear them, then set them back at the end.